src/main/java/com/sourcegraph/cody/chat/CodeEditorFactory.java

```diff
@@ -139,14 +146,70 @@ public void mouseExited(@NotNull EditorMouseEvent event) {
             codeEditorButtons.setVisible(false);
           }
         };
-
     editor.addEditorMouseMotionListener(editorMouseMotionListener);
     editor.addEditorMouseListener(editorMouseListener);
     CodeEditorPart codeEditorPart = new CodeEditorPart(layeredEditorPane, editor);
     codeEditorPart.updateLanguage(language);
     return codeEditorPart;
   }
 
+  @Nullable private TransparentButton attributionIndicator() {
+    AttributionButton guardrailsButton = new AttributionButton("Attribution search running...");
+    guardrailsButton.setEnabled(false);
+    // TODO ACTUALLY GET ATTRIBUTION
+    CompletableFuture.runAsync(() -> {
+      try {
+        Thread.sleep(5000);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      }
+    }).thenAccept((v) -> {
+      guardrailsButton.setText("No attribution");
```

Comment:
It's not allowed to do UI changes on the background threads (one of the IJ development pitfals).
You need to switch to dispatch thread (also called EDT) to do that operation:

```kotlin
ApplicationManager.getApplication().invokeLater {
    guardrailsButton.setText("No attribution");
}
```

Very useful read on that topics (kind of 'must have' for IJ devs in my opinion):
https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html

Reply:
I've left an `invokeLater` there. In a follow-up PR, I'll actually implement it with [Olaf's advice](https://sourcegraph.slack.com/archives/C059N5FRYG3/p1705655269541249?thread_ts=1705588365.312009&cid=C059N5FRYG3).